internal/templates/render/render.go

```diff
@@ -0,0 +1,177 @@
+			`"mylocaltarball.tgz" is for a template not in git but present on the local filesystem. ` +
+			`"http://example.com/myremotetarball.tgz" os for a non-Git template in a remote tarball.`,
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "spec",
+		Aliases: []string{"s"},
+		Example: "path/to/spec.yaml",
+		Target:  &c.spec,
+		Default: "./spec.yaml",
+		Usage:   "the path of the .yaml file within the template directory that specifies how the template is rendered.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "dest",
+		Aliases: []string{"d"},
+		Example: "/my/git/dir",
+		Target:  &c.dest,
+		Default: ".",
+		Usage:   "the target directory in which to write the output files.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "git-protocol",
+		Aliases: []string{"p"},
+		Example: "https",
+		Target:  &c.gitProtocol,
+		Usage:   "either ssh or https, the protocol for connecting to GitHub.",
+	})
+	sect.StringMapVar(&cli.StringMapVar{
+		Name:    "input",
+		Aliases: []string{"i"},
+		Example: "foo=bar",
+		Target:  &c.inputs,
+		Usage:   "key=val pairs of template values; may be repeated.",
+	})
+	sect.StringVar(&cli.StringVar{
+		Name:    "log-level",
+		Aliases: []string{"l"},
+		Example: "info",
+		Default: "warning",
+		Target:  &c.logLevel,
+		Usage:   "how verbose to log; any of debug|info|warning|error.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "allow-non-git-dest",
+		Aliases: []string{"a"},
+		Target:  &c.allowNonGitDest,
+		Usage:   "skip the sanity checks that the dest directory is a git repo.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "force-overwrite",
+		Aliases: []string{"o"},
+		Target:  &c.forceOverwrite,
+		Usage:   "if an output file already exists in the destination, overwrite it instead of failing.",
+	})
+	sect.BoolVar(&cli.BoolVar{
+		Name:    "keep-temp-dirs-on-failure",
+		Aliases: []string{"k"},
+		Target:  &c.keepTempDirs,
+		Usage:   "if there is an error, preserve the temp directories instead of deleting them normally.",
+	})
+
+	return set
+}
+
+func (c *Command) parseFlags(args []string) error {
+	if err := c.Flags().Parse(args); err != nil {
+		return fmt.Errorf("failed to parse flags: %w", err)
+	}
+
+	if c.template == "" {
+		return fmt.Errorf("-template is required")
+	}
+	if c.dest == "" {
+		return fmt.Errorf("-dest is required")
+	}
+
+	if c.allowNonGitDest {
+		ok, err := isGitRepo(c.dest)
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return fmt.Errorf("the provided -dest dir is not a git repo; if this is intentional, use --allow-non-git-dest")
+		}
+	}
+
+	if c.template == "" {
+		return fmt.Errorf("-template is required")
+	}
+
+	return nil
+}
+
+func (c *Command) Run(ctx context.Context, args []string) error {
+	if err := c.parseFlags(args); err != nil {
+		return err
+	}
+	return fmt.Errorf("stub")
+}
+
+func isGitRepo(dir string) (bool, error) {
+	path := filepath.Join(dir, ".git")
+	_, err := os.Stat(path)
+	if err == nil {
+		return true, nil
```

Comment:
Nit: prefer cuddling
```suggestion
	if _, err := os.Stat(path); err == nil {
		return true, nil
```

Reply:
This got rewritten for other reasons, resolving